components/EventForm.tsx

```diff
@@ -0,0 +1,59 @@
+'use client';
+
+import { useState } from "react";
+import { Event } from "@prisma/client";
+
+export default function EventForm({ postEvent, userId }) {
+	const [title, setTitle] = useState('');
+	const [location, setLocation] = useState('');
+	const [accessStart, setAccessStart] = useState(Date());
+	const [accessEnd, setAccessEnd] = useState(Date()); 
+
+	const temp_inviteLink = 'www.templink.com/until/we/can/gen/ourselves';
+	const temp_verifierCode = 'abc123';
+
+	const event: Event = {
+		title,
+		location,
+		accessStart: new Date(accessStart),
+		accessEnd: new Date(accessEnd),
+		inviteLink: temp_inviteLink,
+		verifierCode: temp_verifierCode,
+		hostId: userId,
+	}
+
+	function handleSubmit(ev: React.FormEvent) {
+		ev.preventDefault();
+		postEvent(event);
+	}
+
+	return (
+		<div>
+			<form onSubmit={handleSubmit}>
```

Comment:
We could use react-hook-form which makes forms more performant. Ref: https://www.react-hook-form.com/

Reply:
fax react-hook-form is actually better and reduces need for useState when handling input